### PR TITLE
feat: Add managedplugin 

### PR DESCRIPTION
**Pull request by yevgenypats**
This adds managedplugin to plugin-pb-go.

In the past we had managedsourceplugin and manageddestinationplugin in the CLI. Now as we move to SDK V3 where we have one protocol for sources and destinations we also merge the `managedplugin` and we move it to plugin-pb-go so SDK and plugin can use this capability as well (for example to spawn plugins to use them as state backends.)

This is also adds a new discovery service that uses `int` instead of strings which is quite unnecessary. 

Also, this removes `specreader` from `specs` as before we had the CLI spec coupled with over-the-wire configuration sent also as spec. Now `specs` only contains over-the-wire structs while the `cli` contains it's own `specs` and `specreader`.

### CODE REVIEW

**Review by erezrokah — APPROVED** [review]

File: managedplugin/download.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+// TODO: we can improve this by just embedding all plugins and last version that exist in different places then
+// the latest
+func getURLLocation(ctx context.Context, org string, name string, version string) (string, error) {
+	urls := []string{
+		fmt.Sprintf("https://github.com/%s/cq-plugin-%s/releases/download/%s/cq-plugin-%s_%s_%s.zip", org, name, version, name, runtime.GOOS, runtime.GOARCH),
+		fmt.Sprintf("https://github.com/%s/cq-source-%s/releases/download/%s/cq-source-%s_%s_%s.zip", org, name, version, name, runtime.GOOS, runtime.GOARCH),
+		fmt.Sprintf("https://github.com/%s/cq-destination-%s/releases/download/%s/cq-destination-%s_%s_%s.zip", org, name, version, name, runtime.GOOS, runtime.GOARCH),
+	}
+	if org == "cloudquery" {
+		urls = append(urls, fmt.Sprintf("https://github.com/cloudquery/cloudquery/releases/download/plugins-plugin-%s-%s/%s_%s_%s.zip", name, version, name, runtime.GOOS, runtime.GOARCH))
```
Comment (hermanschaaf) [inline managedplugin/download.go:38]:
```suggestion
		urls = append(urls, fmt.Sprintf("https://github.com/cloudquery/cloudquery/releases/download/plugins-%s-%s/%s_%s_%s.zip", name, version, name, runtime.GOOS, runtime.GOARCH))
```

I think if we moved to this structure (which IMO we shouldn't in this migration, but maybe later), we'd want all plugins to live directly under `plugins/` without the additional `plugins/plugin` layer

**Review by hermanschaaf — APPROVED** [review]
LGTM apart from the few comments which are relatively minor. We should make a note in the release notes that this update will cause plugin binaries to be downloaded again if they were cached before, but I think that shouldn't be an issue for most users.

File: managedplugin/download.go (hunk 1 above)
Comment (yevgenypats) [reply managedplugin/download.go]: agree. I kept the structure the same for now and also removed the additional `HEAD` as we don't have any plugins there for now.

File: managedplugin/plugin.go [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+			return nil, err
+		}
+	case RegistryGithub:
+		pathSplit := strings.Split(config.Path, "/")
+		if len(pathSplit) != 2 {
+			return nil, fmt.Errorf("invalid github plugin path: %s. format should be owner/repo", config.Path)
+		}
+		org, name := pathSplit[0], pathSplit[1]
+		c.LocalPath = filepath.Join(c.directory, "plugins", "github.com", org, name, config.Version, "plugin")
+		c.LocalPath = WithBinarySuffix(c.LocalPath)
```
[the remark this answers is not in the document]
Comment (yevgenypats) [reply managedplugin/plugin.go:155]: Reverted it to use the old plugin path so it wont introduce a braking change

**Pull request merged**